controller/sharding/sharding.go

```diff
@@ -162,3 +186,156 @@ func createClusterIndexByClusterIdMap(db db.ArgoDB) map[string]int {
 	}
 	return clusterIndexedByClusterId
 }
+
+/*
+ * GetShardFromConfigMap finds the shard number from the shard mapping configmap. If the shard mapping configmap does not exist,
+ * the function creates the shard mapping configmap. If the shard number is set as an environment variable for the application controller pod,
+ * we use the shard number to update the mapping.
+ */
+func GetShardFromConfigMap(kubeClient *kubernetes.Clientset, settingsMgr *settings.SettingsManager, replicas, shard int) (int, error) {
+
+	hostname, err := osHostnameFunction()
+	if err != nil {
+		return -1, err
+	}
+
+	// fetch the shard mapping configMap
+	shardMappingCM, err := kubeClient.CoreV1().ConfigMaps(settingsMgr.GetNamespace()).Get(context.Background(), common.ArgoCDAppControllerShardConfigMapName, metav1.GetOptions{})
+
+	if err != nil {
+		log.Infof("shard mapping configmap %s not found. Creating default shard mapping configmap.", common.ArgoCDAppControllerShardConfigMapName)
+
+		// if the shard is not set as an environment variable, set the default value of shard to 0 for generating default CM
+		if shard == -1 {
```

Comment:
What shard == -1 means exactly? I think that this is an importat info to add in the godoc of this function.

Reply:
So, the users do have a capability to set the shard number for a specific controller today. The function takes the shard number from the environment variable (default value -1, if not set) and passes it to this function. So, if the shard value passed to this function is `-1`, that is, the shard was not set as an environment variable, we default the shard number to 0 for computing the default config map.

Reply:
Done